src/browser/ui/dom/components/ReactDOMInput.js

```diff
@@ -115,6 +138,24 @@ var ReactDOMInput = ReactClass.createClass({
   _handleChange: function(event) {
     var returnValue;
     var onChange = LinkedValueUtils.getOnChange(this.props);
+
+    // IE 10+ fire input events when setting/unsetting a placeholder
+    // we guard against it by checking if the next and
+    // last values are both empty and bailing out of the change
+    // https://github.com/facebook/react/issues/3484
+    if (isIEInputEvent(event)) {
+      var controlledValue = LinkedValueUtils.getValue(this.props);
+      var lastValue = controlledValue != null ?
```

Comment:
oof I hadn't thought of that, you mean if the user provides a `value` props but then manually changes the DOM node value like `findDOMNode(input).value = "can't touch this!"`? I want to say in those cases the "right" thing to do is use the `value` prop as the last value, because the props existence indicates an intent to always handle the value declaratively.